examples/http/client.cpp

```diff
@@ -0,0 +1,108 @@
+// Simple HTTP client that prints the response
+
+#include "caf/net/http/client.hpp"
+
+#include "caf/net/http/with.hpp"
+#include "caf/net/ip.hpp"
+#include "caf/net/middleman.hpp"
+#include "caf/net/octet_stream/transport.hpp"
+#include "caf/net/tcp_stream_socket.hpp"
+
+#include "caf/actor_system.hpp"
+#include "caf/actor_system_config.hpp"
+#include "caf/caf_main.hpp"
+#include "caf/detail/latch.hpp"
+#include "caf/event_based_actor.hpp"
+#include "caf/ipv4_address.hpp"
+#include "caf/scheduled_actor/flow.hpp"
+#include "caf/span.hpp"
+#include "caf/uuid.hpp"
+
+#include <cassert>
+#include <iostream>
+#include <memory>
+#include <utility>
+
+using namespace std::literals;
+
+namespace http = caf::net::http;
+namespace ssl = caf::net::ssl;
+using namespace caf;
+
+// -- constants ----------------------------------------------------------------
+
+constexpr auto default_method = http::method::get;
+
+// -- configuration setup ------------------------------------------------------
+struct config : actor_system_config {
+  config() {
+    opt_group{custom_options_, "global"}
+      .add<http::method>("method,m", "HTTP method to use")
+      .add<std::string>("payload,p", "Optional payload to send");
+    opt_group{custom_options_, "tls"} //
+      .add<bool>("enable", "enables encryption via TLS")
+      .add<std::string>("ca-file", "CA file for trusted servers");
+  }
+
+  caf::settings dump_content() const override {
+    auto result = actor_system_config::dump_content();
+    caf::put_missing(result, "method", default_method);
+    return result;
+  }
+};
+
+int caf_main(caf::actor_system& sys, const config& cfg) {
+  // Get URI from config (positional argument).
+  if (cfg.remainder.size() != 1) {
+    std::cerr << "*** expected mandatory positional argument: URL" << std::endl;
+    return EXIT_FAILURE;
+  }
+  uri resource;
+  if (auto err = parse(cfg.remainder[0], resource)) {
+    std::cerr << "*** failed to parse URI" << std::endl;
+    return EXIT_FAILURE;
+  }
+  auto use_ssl = caf::get_or(cfg, "tls.enable", false);
+  auto ca_file = caf::get_as<std::string>(cfg, "tls.ca-file");
+  auto method = caf::get_or(cfg, "method", default_method);
+  auto payload = caf::get_or(cfg, "payload", ""sv);
+  auto result
+    = http::with(sys)
+        // Optionally enable TLS.
+        .context(ssl::context::enable(use_ssl)
+                   .and_then(ssl::emplace_client(ssl::tls::v1_2))
+                   .and_then(ssl::load_verify_file_if(ca_file)))
```

Comment:
Not happy with this. The factory should know when to initialize TLS by itself (depending on whether the scheme is HTTP or HTTPS). As a user, I'd want something like this:

```
http::with(sys)
.tls_verify_file(caf::get_as<std::string>(cfg, "tls.ca-file"))
.connect(resource)
...
```

With `tls_verify_file` having no effect if:
- `caf::get_as` does not return a value
- the scheme is HTTP